peps/pep-9999.rst

```diff
@@ -0,0 +1,233 @@
+Status: Draft 
+Type: Standards Track
+Content-Type: text/x-rst
+Created: 28-Oct-2023
+Post-History: 14-Oct-2023
+
+Abstract
+==================
+
+This PEP proposes introducing syntatic sugar ``f(x=)`` for the common
+pattern where a named argument is the same as the name of the variable
+corresponding to its value ``f(x=x)``.
+
+Motivation
+==================
+Keyword argument syntax can become needlessly repetitive and verbose.
+
+Consider the following call:
+::
+
+    my_function(
+      my_first_variable=my_first_variable,
+      my_second_variable=my_second_variable,
+      my_third_variable=my_third_variable,
+    )
+
+The case of a keyword argument name matching the variable name of its value is prevalent among all major Python libraries. This verbosity and redundancy discourages use of named arguments and reduces readability by increasing visual noise.
+
+Rationale
+==================
+There are two ways to invoke a function with arguments: by position and by keyword. Keyword arguments confer many benefits by being explicit, thus increasing readability and minimising the risk of inadvertent transposition. On the flipside, positional arguments are often used simply to minimise verbosity and visual noise.
+
+We contend that a simple syntactic sugar used to simplify this common pattern which would confer numerous benefits:
+
+Encourages use of named variables
+------------------------------------
+This syntax would encourage the use of named variables, thereby increasing readability (*explicit is better than implicit*) and reducing bugs from argument transposition.
+
+Reduces verbosity
+------------------------------------
+By minimising visual noise and in some cases lines of code, we can increase readability (*readability counts*).
+
+Encourages consistent variable names
+------------------------------------
+A common problem is that semantically identical variables have different names depending on their contexts. This syntax would encourage authors to use the same variable name when calling a function as the argument name, which would increase consistency of variable names used and hence also *readability*.
+
+Naturally extends to dictionaries
+------------------------------------
+This syntax offers a natural way to support this syntactic sugar for dictionary construction where a similar pattern occurs (dictionary keys are identical the names of the variables assigned as their values), ``{"x": x, "y": y}`` or ``dict(x=x, y=y)``. With this feature, this can now also be trivially written as ``dict(x=, y=)``. Whether to further support similar syntax in dictionary literals is an open question out of the scope of this PEP.
+
+Specification
+==================
+
+The proposed syntactic sugar is very simple. Instead of:
+::
+
+    my_function(param1=param1, param2=param2)
+
+You may instead write:
+::
+
+    my_function(param1=, param2=)
+
+
+Precedence
```

Comment:
In most proposals, "precedence" refers to the order in which operators are evaluated. Maybe "Prior Art"? Great example in [PEP 515](https://peps.python.org/pep-0515/#prior-art) and several other places it's been used too.